### PR TITLE
Ingress

### PR DESCRIPTION
**Pull request by EwanNoble**

### CODE REVIEW

File: src/SFA.DAS.ToolService.Core/Services/ApplicationService.cs [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+        public async Task<List<Application>> GetApplicationsForRole(string[] roles)
+        {
+            var applications = new List<Application>();
+            var publicApplications = await _applicationRepository.GetPublicApplications();
+            applications.AddRange(publicApplications);
+
+            foreach (var role in roles)
+            {
+                var roleId = await _applicationRepository.GetRoleId(role);
+                var app = await _applicationRepository.GetApplicationsInRole(roleId);
```
Comment (nbowes24) [inline src/SFA.DAS.ToolService.Core/Services/ApplicationService.cs]: Could this be joined? Kinda like this? Add a virtual property Role.

        public async Task<List<Application>> GetAppsByRoleName(string name)
        {        
            var result = await _toolServiceDbContext.ApplicationRole
                .Where(c => c.Role.Name.Equals(name))
                .Select(c => c.Application).ToListAsync();

            return result;
        }

File: src/SFA.DAS.ToolService.Core/Entities/ApplicationRole.cs [hunk 2]
```diff
@@ -0,0 +1,11 @@
+namespace SFA.DAS.ToolService.Core.Entities
+{
+    public class ApplicationRole
+    {
+        public int Id { get; set; }
+        public int ApplicationId { get; set; }
+        public int RoleId { get; set; }
+
+        public virtual Application Application { get; set; }
+    }
```
Comment (nbowes24) [inline src/SFA.DAS.ToolService.Core/Entities/ApplicationRole.cs]: If wanting to do the join

public virtual Role Role { get; set; }

File: src/SFA.DAS.ToolService.Authentication/ServiceCollectionExtensions/AddAuthenticationProviders.cs [hunk 3]
```diff
@@ -10,70 +10,6 @@ namespace SFA.DAS.ToolService.Authentication.ServiceCollectionExtensions
 {
     public static class ServiceCollectionExtensions
```
[the remark this answers is not in the document]
Comment (chelnak) [reply src/SFA.DAS.ToolService.Authentication/ServiceCollectionExtensions/AddAuthenticationProviders.cs]: It's moved to the web project

**Comment by chelnak** [comment]
@nbowes24 Changes implemented and it appears to work in dev

**Pull request merged**